tests/memory-usage.rs

```diff
@@ -0,0 +1,126 @@
+#[salsa::input]
+struct MyInput {
+    field: u32,
+}
+
+#[salsa::tracked]
+struct MyTracked<'db> {
+    field: u32,
+}
+
+#[salsa::interned]
+struct MyInterned<'db> {
+    field: u32,
+}
+
+#[salsa::tracked]
+fn input_to_interned<'db>(db: &'db dyn salsa::Database, input: MyInput) -> MyInterned<'db> {
+    MyInterned::new(db, input.field(db))
+}
+
+#[salsa::tracked]
+fn input_to_tracked<'db>(db: &'db dyn salsa::Database, input: MyInput) -> MyTracked<'db> {
+    MyTracked::new(db, input.field(db))
+}
+
+#[salsa::tracked]
+fn input_to_tracked_tuple<'db>(
+    db: &'db dyn salsa::Database,
+    input: MyInput,
+) -> (MyTracked<'db>, MyTracked<'db>) {
+    (
+        MyTracked::new(db, input.field(db)),
+        MyTracked::new(db, input.field(db)),
+    )
+}
+
+#[test]
+fn test() {
+    let db = salsa::DatabaseImpl::new();
+
+    let input1 = MyInput::new(&db, 1);
+    let input2 = MyInput::new(&db, 2);
+    let input3 = MyInput::new(&db, 3);
+
+    let _tracked1 = input_to_tracked(&db, input1);
+    let _tracked2 = input_to_tracked(&db, input2);
+
+    let _tracked_tuple = input_to_tracked_tuple(&db, input1);
+
+    let _interned1 = input_to_interned(&db, input1);
+    let _interned2 = input_to_interned(&db, input2);
+    let _interned3 = input_to_interned(&db, input3);
+
+    let structs_info = <dyn salsa::Database>::structs_info(&db);
+
+    assert_eq!(
+        format!("{structs_info:#?}"),
+        r#"[
+    IngredientInfo {
```

Comment:
This will probably get annoying to maintain. Ideally we'd have some form of snapshot testing here.

Reply:
You can use `expect!`

https://github.com/MichaReiser/salsa/blob/87a730fccf07add73aabc6f2c842d92fd2bfe466/tests/accumulated_backdate.rs#L65-L71